pkgs/by-name/ta/taisei/package.nix

```
@@ -0,0 +1,104 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  # Build depends
+  docutils,
+  meson,
+  ninja,
+  pkg-config,
+  python3,
+  python312Packages,
+  ensureNewerSourcesForZipFilesHook,
+  # Runtime depends
+  glfw,
+  SDL2,
+  SDL2_mixer,
+  cglm,
+  freetype,
+  libpng,
+  libwebp,
+  libzip,
+  zlib,
+  zstd,
+  opusfile,
+  openssl,
+  gamemode,
+  shaderc,
+  spirv-cross,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "taisei";
+  version = "1.4.2";
+
+  src = fetchFromGitHub {
+    owner = "taisei-project";
+    repo = "taisei";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-rThLz8o6IYhIBUc0b1sAQi2aF28btajcM1ScTv+qn6c=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    docutils
+    meson
+    ninja
+    pkg-config
+    python3
+    python312Packages.zstandard
+    ensureNewerSourcesForZipFilesHook
+    shaderc
+  ];
+
+  buildInputs = [
+    glfw
+    SDL2
+    SDL2_mixer
+    cglm
+    freetype
+    libpng
+    libwebp
+    libzip
+    zlib
+    zstd
+    opusfile
+    openssl
+    gamemode
+    spirv-cross
+  ];
+
+  mesonFlags = [
+    (lib.mesonBool "b_lto" false)
+    (lib.mesonEnable "install_macos_bundle" false)
+    (lib.mesonEnable "install_relocatable" false)
+    (lib.mesonEnable "shader_transpiler" false)
+  ];
```

```suggestion
    (lib.mesonEnable "gamemode" stdenv.hostPlatform.isLinux)
  ];
```

by adding this and also putting the `gamemode` dependency behind `lib.optionals stdenv.hostPlatform.isLinux` the Darwin build will be fixed.

You could even add a `gamemodeSupport ? stdenv.hostPlatform.isLinux` at the top of the file and use `gamemodeSupport` as the condition. This way even on Linux it can be turned off.

---

That is indeed better, I will do that.

As for the darwin build, I'm a bit conservative to unmark it as `broken` as I do not have a darwin system to confirm if it does work. Should I just go forward and see if the CI manage to build the package?

---

I have a local VM set up to test darwin stuff, but before that I just used to rely on OfBorg results (though that might be a bit too much waiting).

I'll check if everything's working for this PR.

If things break on darwin in the future, you can just mark the package as broken again if you don't want to bother with it.

---

You can force OfBorg to start building earlier by doing `@`ofborg build some-package

(I used `@` instead of @ because I didn't want to actually trigger it)

---

Good to know. I might look into setting up a darwin VM in the future for such use case.